pages/upgrade-guide/4.0.md

```diff
@@ -0,0 +1,186 @@
+
+### CSS
+
+#### Breaking change: `px` no longer automatically added to most unitless values
+
+jQuery 4.0 switches strategies when it comes to setting CSS properties that require units. Previously, jQuery would automatically add `px` to unitless values, with some exceptions. This behavior has been removed in jQuery 4.0.
+
+Instead, jQuery will now only add `px` to a limited set of properties that are known to require it. This change may affect existing code that relies on the old behavior.
+
+### Data
+
+#### Important fix: `Object.prototype` pollution
+
+jQuery 4.0 includes a fix to ensure event and data keys matching `Object.prototype` properties are supported without overriding the native properties. This change is not expected to affect existing code unless it specifically used native property names as event or data keys, which may have caused unexpected behavior anyway.
+
+### Deferred
+
+#### Breaking change: Removed jQuery.Deferred.getStackHook
+
+This has been renamed since 3.7.0 to jQuery.Deferred.getErrorHook. It is used to pass the original error to jQuery.Deferred.exceptionHook. Previously, we recommended passing the stack but that didn't play well with source maps.
+
+### Event
+
+#### Breaking change: Stop shimming focusin & focusout events
+
+jQuery 4.0 no longer shims the `focusin` and `focusout` events. This only affects code that relies on the order of these events as they relate to `focus` and `blur`. This change is expected to affect very few users.
+
+#### Breaking change: Avoid collisions between jQuery.event.special & Object.prototype
+
+This may affect code calling `hasOwnProperty` on `jQuery.event.special`, which should  be rare.
+
+### Manipulation
+
+#### Important feature: Add support for Trusted Types
+
+jQuery 4.0 adds support for [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API) for all manipulation methods. This change is not expected to affect existing code, but it is an important feature for security.
+
+#### Potentially breaking change: Avoid concatenating strings in buildFragment
+
+This change was needed in order to add support for [Trusted Types](https://developer.mozilla.org/en-US/docs/Web/API/Trusted_Types_API). It's not entirely clear what real use cases might be affected by this change, but we're listing it here just in case.
+
+#### Important fix: Don't remove HTML comments from scripts
+
+Browsers can now handle this without jQuery's help. This change is not expected to affect existing code, but is an important fix because jQuery's logic removing those comments was not correctt in 100% of cases.
```

Comment:
This was done on `3.x-stable` as well (released in `3.6.1`) and we don't treat it as breaking. What was not done in 3.x but done on `main` was no longer removing CDATA section markers either.

Compare:
* 3.x version: https://github.com/jquery/jquery/commit/924b515dd3f33fff3e5122408fcccecc2d775425
* 4.x version: https://github.com/jquery/jquery/commit/2f8f39e457c32c454c50545b0fdaa1d7a4a2f8bd